modules/Hoot/ui/sidebar/advancedOpts.js

```diff
@@ -93,8 +117,77 @@ export default class AdvancedOpts {
                             showingOpts.push(a.name);
                         }
                     } );
+                // hide all fav opt buttons
+                d3.select('#saveFav').classed('hidden', true);
+                d3.select('#updateFav').classed('hidden', true);
+                d3.select('#deleteFav').classed('hidden', true);
+                d3.select('#conflateType').property('value', 'Reference');
                 this.createGroups(this.advancedOptions, showingOpts);
             });
+
+            favoritesBar
+                .append( 'div' )
+                .classed( 'fav-button-placement', true )
+                .append( 'button' )
+                .classed( 'advanced-opts-reset button secondary strong hidden', true )
+                .attr( 'id', 'saveFav')
+                .text( 'Save Favorite' )
+                .on('click', async item => {
+
+                    Hoot.getAllUsers();
+
+                    let currentFavorites = Hoot.config.users[Hoot.user().id].members;
+
+                    this.saveOpts = new SaveFavoriteOpt(currentFavorites).render();
+
+                    this.saveOpts.saveOpt[0].data = this.savingFavoriteOpts();
+
+                    Hoot.events.once('modal-closed', () => delete this.saveOpts);
+
+                    return this;
+
+                });
+
+            favoritesBar
+                .append( 'div' )
+                .classed( 'fav-button-placement', true )
+                .append( 'button' )
+                .classed( 'advanced-opts-reset button secondary strong hidden', true )
+                .attr( 'id', 'updateFav')
+                .text( 'Update Favorite' )
+                .on( 'click', async function() {
+
+                    let activeFavorite = d3.select( '#conflateType' ).property( 'value' );
+
+                    let optUpdate = instance.favoriteOptions.filter( opt => opt.name === activeFavorite );
+
+                    let toUpdate = instance.updateFavoriteOpt( optUpdate );
+
+                    let updateOpt = {
+                        name: activeFavorite,
+                        members: {
+                            members: toUpdate,
+                            name: activeFavorite,
+                            label: activeFavorite
+                        }
+                    };
+
+                    await Hoot.api.saveFavoriteOpts( updateOpt );
+
+                    await Hoot.getAllUsers();
+                } );
+
+            favoritesBar
+                .append( 'div' )
+                .classed( 'fav-button-placement', true )
+                .append( 'button' )
+                .classed( 'advanced-opts-reset button secondary strong hidden', true )
+                .attr( 'id', 'deleteFav')
+                .text( 'Delete Favorite' )
+                .on( 'click', function() {
+                    this.deleteOpts = new DeleteFavoriteOpt().handleSubmit();
```

Comment:
can be replaced with just `new DeleteFavoriteOpt().handleSubmit();` since this.deleteOpts doesn't seem to be used